geo: correctly handle negative SRIDs

otan opened the pull request:
Refs: https://github.com/cockroachdb/django-cockroachdb/issues/148

Negative SRIDs in PostGIS bounce around and become 0. Apply the same
logic to parsing as well as typmod logic.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50559)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50559#-:-MAXm-_j15xLVgtxHwtg:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan and @sumeerbhola)

___
*[pkg/geo/parse.go, line 67 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50559#-MAXjp3vEy7Ln6sjg2Rb:-MAXjp3w-uQ1FKZO6YHm:b-xve6rv) ([raw file](https://github.com/cockroachdb/cockroach/blob/c715be85662e974b2cb8c6012b3224209688a144/pkg/geo/parse.go#L67)):*
> ```Go
> 		return geopb.SpatialObject{}, err
> 	}
> 	// TODO(otan): check SRID is valid against spatial_ref_sys.
> ```

was this TODO referring to the validity of `t.SRID()` and was this just a stale TODO that is being removed?
___
*[pkg/sql/logictest/testdata/logic_test/geospatial, line 31 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50559#-MAXkxoIAPMlgjsD417n:-MAXkxoIAPMlgjsD417o:by9z29e) ([raw file](https://github.com/cockroachdb/cockroach/blob/fae410d90ad5975fe8b5ee958d4d6b857f53ecc8/pkg/sql/logictest/testdata/logic_test/geospatial#L31)):*
> ```
> statement ok
> CREATE TABLE geo_table_negative_values(
>   a geometry(geometry, -1),
> ```

even a negative SRID value in the column type should be "fixed" instead of rejected?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50559)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[pkg/geo/parse.go, line 67 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50559#-MAXjp3vEy7Ln6sjg2Rb:-MAXnhtu9HmvEa2cp-8M:bj7f58z) ([raw file](https://github.com/cockroachdb/cockroach/blob/c715be85662e974b2cb8c6012b3224209688a144/pkg/geo/parse.go#L67)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

was this TODO referring to the validity of `t.SRID()` and was this just a stale TODO that is being removed?
</blockquote></details>

Stale now :) we do this in NewGeometry.
___
*[pkg/sql/logictest/testdata/logic_test/geospatial, line 31 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50559#-MAXkxoIAPMlgjsD417n:-MAXnjvV2njX3jjFt4qL:b-cph3aj) ([raw file](https://github.com/cockroachdb/cockroach/blob/fae410d90ad5975fe8b5ee958d4d6b857f53ecc8/pkg/sql/logictest/testdata/logic_test/geospatial#L31)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

even a negative SRID value in the column type should be "fixed" instead of rejected?
</blockquote></details>

Yep, that's how PostGIS does it.


<!-- Sent from Reviewable.io -->

otan committed:
geo: correctly handle negative SRIDs

Negative SRIDs in PostGIS bounce around and become 0. Apply the same
logic to parsing as well as typmod logic.

Release note: None

otan commented:
bors r=sumeerbhola
